src/cmd/build.js

```diff
@@ -62,3 +69,35 @@ export function prepareBuildDir(buildDir: string): Promise {
     }))
     .then(() => buildDir);
 }
+
+
+/*
+ * Allows or ignores files when creating a ZIP archive.
+ */
+export class FileFilter {
+  filesToIgnore: Array<string>;
+
+  constructor({filesToIgnore}: Object = {}) {
+    this.filesToIgnore = filesToIgnore || [
```

Comment:
Good points! I fixed the first two. As for concatenating filter overrides, I left it as is. I think this will become more clear once we implement customizations -- I don't want to make it too flexible before then. The feature I added to override all default filter rules will probably still be necessary since it is valid to include nested XPIs inside an XPI. Some very large add-ons do this.